src/main/java/seedu/address/logic/commands/SetCommand.java

```diff
@@ -0,0 +1,142 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ingredient.Amount;
+import seedu.address.model.ingredient.Ingredient;
+import seedu.address.model.ingredient.IngredientName;
+
+/**
+ * Set the level of one specific ingredient to a specific level.
+ */
+public class SetCommand extends Command {
+
+    public static final String COMMAND_WORD = "set";
+
+    public static final String MESSAGE_ARGUMENTS = "Ingredient: %s, Amount: %2$s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :set the ingredient in tCheck."
+            + "Parameters: "
+            + PREFIX_INGREDIENT
+            + PREFIX_AMOUNT
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INGREDIENT + "Milk "
+            + PREFIX_AMOUNT + "90 ";
+
+    public static final String MESSAGE_SUCCESS = "Ingredient set: %1$s";
+    public static final String MESSAGE_NO_CHANGE = "Ingredient level already set.";
+    private final Ingredient target;
+    private final SetIngredientDescriptor setIngredientDescriptor;
+
+    /**
+     * Constructs a set command with the given ingredient name and amount.
+     */
+    public SetCommand(Ingredient target, SetIngredientDescriptor setIngredientDescriptor) {
+        requireNonNull(target);
+        requireNonNull(setIngredientDescriptor);
```

Comment:
Perhaps you can use `requireAllNonNull(...)` instead.

Reply:
I followed from the EditCommand, so I guess this is fine ?